### ISSUE TITLE
LWscale positions chroma wrong.

### ISSUE THREAD

**Issue by mightyhuhn** [Nevcairiel/LAVFilters#625]
steps to reproduce.
disable everything except p016 in lavfilter in mpc-hc
take a PNG or webp (jpg may use a different source filter)

https://slow.pics/c/UHEltdhq
![Gundam5](https://github.com/user-attachments/assets/5bbbac63-c5d4-4a42-89cf-3dda8852138e)

madVR does the same but i will not rule a out a general error with video renderers.
for technical both assume mpeg2 but it seem it is moved by an entire pixel to the left.

the important of this bug report is close to none existent.

**Comment by Nevcairiel** [comment]
Whats LWscale and what does it have to do with LAV?

**Comment by mightyhuhn** [comment]
good point whats the name of the lib that does the subsampling lav filter?

**Comment by kasper93** [comment]
> good point whats the name of the lib that does the subsampling lav filter?

you mean swscale? Anyway, send sample.

**Comment by kasper93** [comment]
> good point whats the name of the lib that does the subsampling lav filter?

do you mean swscale? Anyway, send input file used to produce this result.

**Comment by mightyhuhn** [comment]
it's the image below the comparison i have shared it. the png.

swscale is possible.

**Comment by Nevcairiel** [comment]
Both swscale API and internals are being reworked right now anyway, also the ability to even subsample in the first place is only offered for testing and/or to ensure any kind of image is output. Its not mean to be fast or high quality.

**Comment by mightyhuhn** [comment]
"the important of this bug report is close to none existent."

so we just rename this to: "chroma placement error when forcing RGB to 420" and call it a day?

i will recheck it when the rework is finished.

**Comment by kasper93** [comment]
> Both swscale API and internals are being reworked right now anyway, also the ability to even subsample in the first place is only offered for testing and/or to ensure any kind of image is output. Its not mean to be fast or high quality.

I agree it can wait for new API, but this would likely be bigger change. For now as a hotfix you could set `VideoChromaSubsampling` to center chroma on output? Maybe smarter renderer like madVR reads this flag. (?)

Or set the chroma offset for swscale, even using old api...

Anyway, not a big deal, the aliasing with bilinear downscaling is insane anyway, so it wouldn't look much better, but indeed chroma location is a thing that is often missing/broken.

**Comment by mightyhuhn** [comment]
madVR ignores the chromaloc always.

mpcVR does it wrong and it looks like they copied from mpv. (for none mpeg2)

as very often i don't need this fixed there other ways to that.

**Comment by mightyhuhn** [comment]
mpcVR and mpv should do this correctly i'm just to stupid to encode.

so yes flagging it as center could be enough and doing center is not a bug it's a valid position.
